impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -243,6 +265,9 @@ public OnHeapStore(final Configuration<K, V> config, final TimeSource timeSource
       });
       if (result == null) {
         getObserver.end(StoreOperationOutcomes.GetOutcome.MISS);
+        if (byteSized && expiredValue.get() != null) {
+          decrementCurrentUsageInBytesIfRequired(getSizeOfKeyWithOffset(key) + getSizeOfValueWithOffset(expiredValue.get().value()) + sizeOfEngine.getCHMOffset());
```

Comment:
Cool idea, storing in valueholder will help in reducing the sizeof overhead, but add a little bit to the storage which should be fine.